api/src/Repository/UserRepository.php

```diff
@@ -221,4 +221,70 @@ public function findActiveLaysInLastYear()
 
         return $result->fetchAllAssociative();
     }
+
+    public function getAllAdminAccounts()
+    {
+        $dql = "SELECT u FROM App\Entity\User u WHERE u.roleName IN('ROLE_ADMIN', 'ROLE_SUPER_ADMIN', 'ROLE_ADMIN_MANAGER')";
+
+        $query = $this
+            ->getEntityManager()
+            ->createQuery($dql);
+
+        return $query->getResult();
+    }
+
+    public function getAllAdminAccountsCreatedButNotActivatedWithin(string $timeframe)
+    {
+        $date = (new DateTime())->modify($timeframe)->format('Y-m-d H:i:s');
+
+        $dql = "SELECT u FROM App\Entity\User u WHERE u.roleName IN('ROLE_ADMIN', 'ROLE_SUPER_ADMIN', 'ROLE_ADMIN_MANAGER')
+                AND u.lastLoggedIn IS NULL
+                AND u.registrationDate < '{$date}' ";
+
+        $query = $this
+            ->getEntityManager()
+            ->createQuery($dql);
+
+        return $query->getResult();
+    }
+
+    public function getAllActivatedAdminAccounts()
+    {
+        $dql = "SELECT u FROM App\Entity\User u WHERE u.roleName IN('ROLE_ADMIN', 'ROLE_SUPER_ADMIN', 'ROLE_ADMIN_MANAGER')
+                AND u.lastLoggedIn IS NOT NULL";
+
+        $query = $this
+            ->getEntityManager()
+            ->createQuery($dql);
+
+        return $query->getResult();
+    }
+
+    public function getAllAdminAccountsNotUsedWithin(string $timeframe)
+    {
+        $date = (new DateTime())->modify($timeframe)->format('Y-m-d H:i:s');
+
+        $dql = "SELECT u FROM App\Entity\User u WHERE u.roleName IN('ROLE_ADMIN', 'ROLE_SUPER_ADMIN', 'ROLE_ADMIN_MANAGER')
+                AND u.lastLoggedIn < '{$date}'  ";
+
+        $query = $this
+            ->getEntityManager()
+            ->createQuery($dql);
+
+        return $query->getResult();
+    }
+
+    public function getAllAdminAccountsUsedWithin(string $timeframe)
+    {
+        $date = (new DateTime())->modify($timeframe)->format('Y-m-d H:i:s');
```

Comment:
You might also want to look at doing the date calculus in the SQL. https://www.postgresql.org/docs/9.1/functions-datetime.html provide a lot of options for doing current_date minus an interval.